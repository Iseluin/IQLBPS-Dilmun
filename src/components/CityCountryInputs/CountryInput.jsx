import React, {useState} from 'react'

export default function CountryInput({errors, children}){
    const [active, setActive] = useState(false)
    function handleActivation(){
        setActive(true)
    }

    return(
        <div data-testid='country-input'>
        <div
          className={[
            'relative border-2 rounded-md mb-2 text-background border-tertiary',
            errors ? 'border-red-500' : '',
          ].join(' ')}
        >
          <input
            className={[
              'outline-none w-full rounded bg-transparent text-sm transition-all duration-200 ease-in-out p-2',
              active ? 'pt-6' : '',
              errors ? 'error' : '',
            ].join(' ')}
            name='country'
            id='country'
            type='text'
            onFocus={handleActivation}
          />

          <label
            className={[
              'absolute top-0 left-0 flex items-center text-tertiary text-opacity-50 p-2 transition-all duration-200 ease-in-out',
              active ? 'text-xs' : 'text-sm',
              errors ? 'error' : '',
            ].join(' ')}
            htmlFor='country'
          >
            {children}
          </label>
        </div>
        {errors && (
          <span className="inline-flex text-sm error">{errors.message}</span>
        )}
      </div>
    )
}