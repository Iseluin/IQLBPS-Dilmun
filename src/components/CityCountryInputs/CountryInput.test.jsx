import { render, screen } from '@testing-library/react';
import CountryInput from './CountryInput';

test('Country input rendered correctly', () => {
    render(<CountryInput/>);
    expect(screen.getByTestId('country-input')).toMatchSnapshot();
})