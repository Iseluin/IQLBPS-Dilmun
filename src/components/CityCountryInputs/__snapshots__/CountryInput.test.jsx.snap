// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country input rendered correctly 1`] = `
<div
  data-testid="country-input"
>
  <div
    class="relative border-2 rounded-md mb-2 text-background border-tertiary "
  >
    <input
      class="outline-none w-full rounded bg-transparent text-sm transition-all duration-200 ease-in-out p-2  "
      id="country"
      name="country"
      type="text"
    />
    <label
      class="absolute top-0 left-0 flex items-center text-tertiary text-opacity-50 p-2 transition-all duration-200 ease-in-out text-sm "
      for="country"
    />
  </div>
</div>
`;
